docs/DeveloperGuide.md

```diff
@@ -343,77 +343,208 @@ b. Require developers to download those libraries manually (this creates extra w
+2. TaskCrusher adds the event to the active list
+3. TaskCrusher displays back the details of the add action.
+
+Use case ends.
+
+**Extensions**
 
-{More to be added}
+2a. The command format is invalid
 
-## Appendix B : Use Cases
+> 2a1. TaskCrusher shows an error message <br>
+  Use case resumes at step 1
+
+2b. The entered time slot is already occupied by another event 
+> 2b1. TaskCrusher notifies the user that the time slot for the event has been already occupied by other event.
+  2b2. User decides whether or not to force the addition. 
+  
+> Use case ends
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
 
-#### Use case: Delete person
+#### <a name = "list"> Use case: Display the active list
 
 **MSS**
 
-1. User requests to list persons
-2. AddressBook shows a list of persons
-3. User requests to delete a specific person in the list
-4. AddressBook deletes the person <br>
-Use case ends.
+1. User requests to list active tasks, active events or both. Additional sort field may be specified.
+2. TaskCrusher displays the active list according to the listing options, assigning each entry an index.
+
+Use case ends
+
+**Extensions**
+
+2a. The command format is invalid
+
+> 2a1. TaskCrusher shows an error message <br>
+> Use case resumes at step 1
+
+2b. The active list is empty.
+> 2b1. TaskCrusher notifies the user that the active list is currently empty.<br/>
+> Use case ends
+
+#### Use case: Mark a task/event as `done`
+
+**MSS**
+
+1. User requests to [display the active list](#list).
+2. User uses the index of the task/event of interest to request to mark it as `done`. Multiple indices can be entered to process more than one entries. 
+3. TaskCrusher moves the task/event from the active list to the expired list.
+
+Use case ends
 
 **Extensions**
 
-2a. The list is empty
+2a. The task/event is not found in the list
+> Use case ends
 
+2b. User decides not to mark a task/event.
 > Use case ends
 
-3a. The given index is invalid
+3a. The index is invalid
+> 3a1. TaskCrusher shows an error message <br>
+Use case resumes at step 2
+<br>
+#### Use case: manage overdue tasks and/or past events
 
-> 3a1. AddressBook shows an error message <br>
-  Use case resumes at step 2
+**MSS**
+
+1. User requests to [display the active list](#list) with the option to list only overdue tasks and/or past events relative to the current time.
+2. For each item in the displayed list, user uses the index to request to mark it as `done`, or dismiss it. Multiple indices may be entered to process more than one items. 
+3. If the user chose to dismiss an item, TaskCrusher prompts the user to enter the reason for it. 
+4. User enters the reason for dismissing a task/event. If user does not need to do so, user does nothing and proceeds.
+5. TaskCrusher moves the task/event from the active list to the expired list.
```

Comment:
OK! over all this particular use case is quite messed up! please clean it up :-)